shopfloor/services/cluster_picking.py

```diff
@@ -696,10 +696,19 @@ def _set_destination_pack_update_quantity(self, move_line, quantity, barcode):
         if not product:
             packaging = search.packaging_from_scan(barcode)
             product = packaging.product_id
-        if product and move_line.product_id == product:
-            quantity += packaging.qty or 1.0
-            response = self._response_for_scan_destination(move_line, qty_done=quantity)
-            return response
+        if product:
+            if move_line.product_id == product:
+                quantity += packaging.qty or 1.0
+                response = self._response_for_scan_destination(
+                    move_line, qty_done=quantity
+                )
+                return response
+            return self._response_for_scan_destination(
+                move_line,
+                message=self.msg_store.product_not_found_in_pickings(),
```

Comment:
I think this is more suitable
```suggestion
                message=self.msg_store.wrong_record(product),
```